src/Button/Button.tsx

```diff
@@ -14,15 +14,15 @@ export type ButtonProps = JSX.IntrinsicElements["button"] & {
 };
 
 const Button: React.FC<ButtonProps> = React.memo(
-    ({ theme, size, block, className, ...props }: ButtonProps): React.ReactElement<void> => {
-        const [processedClassName, setProcessedClassName] = React.useState<string>("btn btn-primary");
+    ({ theme = "primary", size, block, ...props }: ButtonProps): React.ReactElement<void> => {
+        const [className, setClassName] = React.useState<string>("btn btn-primary");
 
         React.useEffect(() => {
-            setProcessedClassName(classnames(["btn", "btn-" + (theme || "primary"), size && "btn-" + size, { "btn-block": block }, className]));
-        }, [size, theme, block, className]);
+            setClassName(classnames("rc", "btn", `btn-${theme}`, { [`btn-${size}`]: size, "btn-block": block }, props.className));
+        }, [size, theme, block, props.className]);
```

Comment:
```suggestion
        }, [size, theme, block, className]);
```

Reply:
Why? I'm setting the value of `className` based on the value of `props.className`

Reply:
It is confusing though. Since you spread props and select your props by their keys . Wouldn't there be collision in this case ?  Will className still be valid in this component when  remove the line below ?
` const [className, setClassName] = React.useState<string>("btn btn-primary");
`

Reply:
Spreading the props in this case just means that I'm constructing a bunch of variables, the ones I specified and the rest are parked inside a variable called `props`. Everything that I didn't specify is accessed using `props.variable`. That's why `className` doesn't exist because it's inside `props`